File: FrontEnd/Eduport-Angular/src/app/PartnershipManagement/Models/Proposal.ts
import { Partnership } from './Partnership';

export interface Proposal {
  idProposal: number;
  proposalName: string;
  proposalDescription: string;
  startDate: string;
  endDate: string;
  plannedAmount: number;
  proposalStatus: string;
  proposalType: string;
  applications?: Partnership[];  // Updated to handle Partnership applications
}

File: backend/micro-partnership/src/main/java/com/partnershipmanagement/Entities/Proposal.java
package com.partnershipmanagement.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.sql.Date;
import java.util.List;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level= AccessLevel.PRIVATE)
@Entity
@Getter
@Setter
public class Proposal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int idProposal;
    String proposalName;
    String proposalDescription;
    Date startDate;
    Date endDate;
    float plannedAmount;
    String proposalStatus;
    String proposalType;
    //@ElementCollection(fetch = FetchType.EAGER)
    //Set<String> agreements;


    // One-to-many relationship: One Proposal can have many Applications
    @OneToMany(mappedBy = "proposals", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Partnership> applications;


}
